src/lib/helpers.ts

```diff
@@ -217,6 +219,18 @@ export const compareAddressesArray = (
 	return new Set(lowerCaseAddresses).size === 1;
 };
 
+export const findAddressByChain = (
+	addresses: IWalletAddress[],
+	chainId: number,
+	chainType?: ChainType,
+) => {
+	return addresses?.find(address =>
+		chainType
```

Comment:
If the EVM chaintype is passed, it won't work correctly

Reply:
Done!